src/main/java/com/github/manolo8/darkbot/Main.java

```diff
@@ -282,7 +282,11 @@ private void tickLogic(boolean running) {
                 setRunning(false);
             } catch (Throwable e) {
                 FeatureDefinition<Module> modDef = featureRegistry.getFeatureDefinition(newModule);
-                if (modDef != null) modDef.getIssues().addWarning("bot.issue.feature.failed_to_tick", IssueHandler.createDescription(e));
+                if (modDef != null &&
+                        modDef.addIssue(e, "bot.issue.feature.failed_to_tick", PluginIssue.Level.WARNING)) {
+                    setModule(new DummyModule());
+                    setRunning(false);
```

Comment:
Just because an exception has happened doesn't mean the whole bot should halt, that's a terrible way to handle the exception.
Simply log it, add it to the list of issues, and hope it's a one-time thing and doesn't keep happening. Back in the day we also made it so the bot tick rate decreased (only tick once per second, delay for 1000ms every time there's an error) in hopes that it fixes itself just due to time passing and in-game state changing, but we no longer do that